automation/generate_ami.sh

```diff
@@ -0,0 +1,181 @@
+#!/bin/bash
+
+set -e
+
+IMAGE="$IMAGE"
+
+AMI_NAME=$AMI_NAME
+AMI_ARCHITECTURE=${AMI_ARCHITECTURE:-x86_64}
+AMI_ROOT_DEVICE_NAME=${AMI_ROOT_DEVICE_NAME:-/dev/sda1}
+AMI_EBS_DELETE_ON_TERMINATION=${AMI_EBS_DELETE_ON_TERMINATION:-true}
+AMI_EBS_VOLUME_SIZE=${AMI_EBS_VOLUME_SIZE:-8}
+AMI_EBS_VOLUME_TYPE=${AMI_EBS_VOLUME_TYPE:-gp2}
+
+BALENA_PRELOAD_APP=${BALENA_PRELOAD_APP:-balena-os-config-preload-amd64}
+IMPORT_SNAPSHOT_TIMEOUT_MINS=${IMPORT_SNAPSHOT_TIMEOUT_MINS:-10}
+BOOT_PARTITION=""
+
+ensure_all_env_variables_are_set() {
+    local env_not_set=""
+    local env_variables="AWS_ACCESS_KEY_ID
+                         AWS_SECRET_ACCESS_KEY
+                         AWS_DEFAULT_REGION
+                         S3_BUCKET
+                         IMAGE
+                         AMI_NAME
+                         BALENARC_BALENA_URL
+                         BALENACLI_TOKEN
+                         PRELOAD_SSH_PUBKEY"
+
+    for env in $env_variables; do
+        [[ -z "${!env}" ]] && echo "ERROR: Missing env variable: $env" && env_not_set=true
+    done
+
+    if [[ "$env_not_set" ]]; then exit 1; fi
+}
+
+
+mount_cleanup() {
+    [[ -n "$BOOT_PARTITION" ]]              && \
+        echo "* Unmounting boot partition"  && \
+        sudo umount $BOOT_PARTITION         && \
```

Comment:
Not a fan of `sudo` on automation scripts as it may block on user input. I would better check that the user executing the script has the permissions to do all the actions it requires and if not log a message and exit.

Reply:
Thanks Alex, that's a typo, is not be necessary